### PR TITLE
Started CVE-2015

### PR DESCRIPTION
**Pull request by Wingfury**
Found a couple promising VCCS and some other info

### CODE REVIEW

File: cves/CVE-2011-2821.yml [hunk 1]
```diff
@@ -32,7 +32,12 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description:
```
Comment (kjs5107) [inline cves/CVE-2011-2821.yml]: Good work! I like your language choice for this description, it makes it fairly easy to understand. The only comment I have is to maybe consider expanding on what you mean by "unexpected behavior" and maybe specifically say what subsystem is affected instead of just saying "the system."

File: cves/CVE-2011-2821.yml [hunk 2]
```diff
@@ -94,10 +101,13 @@ discovered:
...
     If there is no evidence as to how this vulnerability was found, then you may
     leave this part blank.
-  answer: 
-  date: 
-  automated: 
-  google: 
+  answer: |
+    The bug was reported by Chris Evans.
+    It seems as though the bug was reported and patched by the library ownwer
+    and the fixes were brought in the the google team.
```
Comment (kjs5107) [inline cves/CVE-2011-2821.yml]: > "...and the fixes were brought in the the google team."

Oops! Extra "the."

File: cves/CVE-2011-2821.yml [hunk 3]
```diff
@@ -116,10 +126,17 @@ interesting_commits:
     interesting in light of the lessons learned from this vulnerability. Any
     emerging themes?
   commits:
-  - commit: 
-    note: 
-  - commit: 
-    note: 
+  - commit: 91d19754d46acd4a639a8b9e31f50f31c78f8c9c
```
Comment (kjs5107) [inline cves/CVE-2011-2821.yml]: Out of curiosity, was this CVE hard to go about researching? Considering it's more upon the fault of an external library, not chromium directly, was it difficult to scour for the information you needed?

File: cves/CVE-2015-1243.yml [hunk 4]
```diff
@@ -32,37 +32,50 @@ description_instructions: |
...
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  When Google first wrote it's DOM implementation the "use-after-free"
+  vulnerability was intoduced. They iterated over a copy of a structure in
+  an Observer class and never checked the conditions of the original before
+  attempting to use it based on the copy. This caused the tab being used to
+  crash.
+  This could allow for a Denial of Service attack or other unspecified impact
```
Comment (kjs5107) [inline cves/CVE-2015-1243.yml]: Do you think integrity could also be compromised here? If the memory that was freed has already been allocated elsewhere, the use of that freed memory may corrupt valid data.

File: cves/CVE-2015-1243.yml [hunk 5]
```diff
@@ -75,9 +88,11 @@ unit_tested:
 
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
-  answer: 
-  code: 
-  fix:
```
Comment (kjs5107) [inline cves/CVE-2015-1243.yml]: Interesting, did they just miss a boundary case in the unit test? Or did they actually add on brand new unit tests entirely?

File: cves/CVE-2015-1243.yml [hunk 6]
```diff
@@ -149,38 +165,44 @@ lessons:
     If you think of another lesson we covered in class that applies here, feel
     free to give it a small name and add one in the same format as these.
   defense_in_depth:
-    applies: 
-    note: 
+    applies: true
+    note: |
+      This would be a really easy but to miss because you would assume
```
Comment (kjs5107) [inline cves/CVE-2015-1243.yml]: > "...easy but to miss..."

Get rid of "but."

File: cves/CVE-2015-1243.yml [hunk 7]
```diff
@@ -149,38 +165,44 @@ lessons:
     If you think of another lesson we covered in class that applies here, feel
     free to give it a small name and add one in the same format as these.
   defense_in_depth:
-    applies: 
-    note: 
+    applies: true
+    note: |
```
Comment (kjs5107) [inline cves/CVE-2015-1243.yml]: Consider changing the wording of this to add clarity, for example:

> This would be a really easy to miss because you would assume nothing about the object would change from the copy you made. No defensive checks were in place to make sure there weren't differences between the copy and the original before it was modified.

I've edited some grammatical errors and changed some phrasing.

File: cves/CVE-2011-2821.yml (hunk 1 above)
Comment (Wingfury) [reply cves/CVE-2011-2821.yml]: its not arbitrary code execution that it could cause but does leave room for a buffer overflow attack.
https://cwe.mitre.org/data/definitions/415.html
I agree that information can be added though, thank you!

File: cves/CVE-2011-2821.yml (hunk 2 above)
Comment (Wingfury) [reply cves/CVE-2011-2821.yml]: fixed

File: cves/CVE-2011-2821.yml (hunk 3 above)
Comment (Wingfury) [reply cves/CVE-2011-2821.yml]: It was hard only to the extent of making sure there wasn't something I was missing. I was really paranoid that I didn't have enough information. Especially given my CVE options were all almost identical (all were related to free and malloc).

File: cves/CVE-2015-1243.yml (hunk 4 above)
Comment (Wingfury) [reply cves/CVE-2015-1243.yml]: Good point!

File: cves/CVE-2015-1243.yml [hunk 8]
```diff
@@ -193,4 +215,14 @@ mistakes:
...
-  answer: 
+  answer: |
+    The coding mistake was made when this code was first created. This code was
+    not modifed again until this bug was found. The time between the bug being
+    found, the cause of the bug and the possible fix was very short (less then a
+    day).
+    This was a simple mistake where the coders assumed that the object they
+    made a copy of couldn't be changed and would remain identical to their
+    copy.
+    This mistake was mostly a coding mistake. The coders made an assumption
```
[the remark this answers is not in the document]
Comment (Wingfury) [reply cves/CVE-2015-1243.yml]: I say mostly because I could see review groups easily looking over this issue. There's a little judgement involved in deciding how many defensive checks become overkill. I feel like the reviewers
either completely missed this as a place where a defensive check needed to be, or decided it wasn't necessary (which wouldn't be a coding mistake but a judgement issue).

File: cves/CVE-2015-1243.yml (hunk 5 above)
Comment (Wingfury) [reply cves/CVE-2015-1243.yml]: Not that I could see immediately after, but it wouldn't surprise me. The unit tests were specific to this one use so would likely not have found any new issues.

File: cves/CVE-2015-1243.yml (hunk 8 above)
[the remark this answers is not in the document]
Comment (Wingfury) [reply cves/CVE-2015-1243.yml]: I think it is important, but I already mention the added unit tests at an earlier point in the yml doc

File: cves/CVE-2011-2821.yml [hunk 9]
```diff
@@ -94,10 +101,13 @@ discovered:
 
     If there is no evidence as to how this vulnerability was found, then you may
     leave this part blank.
-  answer: 
-  date: 
-  automated: 
-  google: 
+  answer: |
+    The bug was reported by Chris Evans.
```
[the remark this answers is not in the document]
Comment (Wingfury) [reply cves/CVE-2011-2821.yml]: Really? Neat! I've actually never heard of him!

File: cves/CVE-2011-2821.yml [hunk 10]
```diff
@@ -194,4 +215,11 @@ mistakes:
...
     engineering industry would find interesting.
-  answer: 
+  answer: |
+    This vulnerability was introduced because of the use of the libxml
+    library. We all use libraries to make our work easier, but have to
+    remember that those libraries can introduce as many vulnerabilities
+    as our own code does.
+    Looking at the changes to the lxml library this looks to be a coding
+    mistake where it was assumed some data structures could never be
+    modified. This turned out to be incorrect.
```
[the remark this answers is not in the document]
Comment (Wingfury) [reply cves/CVE-2011-2821.yml]: I think the lxml library isn't even used anymore, so I am sure as they faced more and more issues with external libraries it really drove them to try and get away from them.

File: cves/CVE-2015-1243.yml [hunk 11]
```diff
@@ -32,37 +32,50 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  When Google first wrote it's DOM implementation the "use-after-free"
```
[the remark this answers is not in the document]
Comment (Wingfury) [reply cves/CVE-2015-1243.yml]: You are totally correct, I'll fix this in the morning.

File: cves/CVE-2015-1243.yml [hunk 12]
```diff
@@ -127,12 +143,12 @@ major_events:
 
     The event doesn't need to be directly related to this vulnerability, rather,
     we want to capture what the development team was dealing with at the time.
-  answer: 
+  answer: There were no major events during the history of this vulnerability.
```
[the remark this answers is not in the document]
Comment (Wingfury) [reply cves/CVE-2015-1243.yml]: I don't think it was discovered specifically then? I didn't see any indication that use after free was newly discovered at the same time as this particular vulnerability.
